src/basic/cgroup-util.c

```diff
@@ -152,6 +152,17 @@ int cg_read_pidref(FILE *f, PidRef *ret, CGroupFlags flags) {
                 r = pidref_set_pid(ret, pid);
                 if (r >= 0)
                         return 1;
+                /* Opening pidfds for non thread group leaders only works from 6.9 onwards. On older kernels
+                 * pidfd_open() fails with EINVAL. Since we might read non thread group leader IDs from
+                 * cgroup.threads and try to open pidfd's for them, let's ignore EINVAL from pidref_set_pid()
+                 * and fallback to using the pid instead. */
+                if (r == -EINVAL) {
```

Comment:
hmm, maybe add a new flag to CGroupFlags called CGROUP_GRACEFUL_EINVAL or so, that enables this, and then OR that into the flags passed to the cg_kill_items() call for the `cgroup.threads` file in cg_kill()? Because that's really the only place we need this, no?

(needs some fixing in cg_kill_items(), since the flags param is currently not propagated to cg_read_pidref(), but really should be)